### PR TITLE
Use eslint instead of jshint

### PR DESCRIPTION
**Pull request by maratfakhreev**
I have add more flexible linter [eslint](http://eslint.org) with initial config. In future this config will be updated to suit our needs, anyway now it supports all backbone base es6/es7 features including gulp tasks.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+  "env": {
+    "browser": false,
+    "es6": true,
+  },
+
+  "rules": {
+    // specify the maximum depth that blocks can be nested
+    "max-depth": [0, 4],
+    // specify the maximum length of a line in your program
+    "max-len": [0, 80, 4],
```
Comment (VladimirMikhailov) [inline .eslintrc]: could you clarify how we can read the config `[0, 80, 4]` - min 0 characters and 80 max? and what for `4` is here? confused a bit , thanks.

File: .eslintrc [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    "max-statements": [0, 10],
+    // disallow use of bitwise operators
+    "no-bitwise": 0,
+    // disallow use of unary operators, ++ and --
+    "no-plusplus": 0,
+
+    // enforce spacing inside array brackets
+    'array-bracket-spacing': 0,
+    // require camel case names
+    'camelcase': [2, {'properties': 'never'}],
```
Comment (VladimirMikhailov) [inline .eslintrc]: can properties be not camelcased? Which properties should never be camelcased? And what 2 is about, please?

```js
@props({
  urlRoot: 'comments'
})
```

File: app/scripts/views/articles/list.js [hunk 3]
```diff
@@ -7,6 +7,6 @@ import { props } from 'scripts/decorators';
 })
 export default class ArticlesListView extends Marionette.CollectionView {
   childViewOptions() {
-    return { page: this.options.page }
+    return { page: this.options.page };
```
Comment (VladimirMikhailov) [inline app/scripts/views/articles/list.js]: > ECMAScript 6 coders nowadays can get rid of nearly all semicolons and remove clutter from their source code. Ralf  S. Engelschall is a strong promoter of reducing source code to its bare minimum. Hence, in his personal opinion
ECMAScript 6 should be coded with as less syntactic sugar as possible and hence without semicolons. 

what do you think about getting rid of unnecessary semicolons?

File: .eslintrc (hunk 2 above)
Comment (VladimirMikhailov) [inline .eslintrc]: 2 - turn the rule on as an error (exit code is 1 when triggered) - is clear, thanks

File: .eslintrc (hunk 1 above)
Comment (VladimirMikhailov) [inline .eslintrc]: ok, got it:
 zero is about rule priority (should it be 2 here -  turn the rule on as an error)
80 is limitation
4 - each tab is 4 characters

File: app/scripts/views/articles/list.js (hunk 3 above)
Comment (maratfakhreev) [inline app/scripts/views/articles/list.js]: I'll think about it and perhaps realise it the following PR's)

**Comment by ArturMinnullin** [comment]
:ok:

**Comment by AmirMingaliev** [comment]
:+1:

**Pull request merged**